Updating information about Bower, Yarn, staging, alpha and beta

dotpointer opened the pull request:
Installation instructions for cosmoz-frontend has changed lately with the introduction of Yarn.

Alpha environment has also been introduced in favor of staging that has been phased out.

The documentation need to follow the changes, and this pull request is intended to start doing so.

I figure there are things missing in this pull request, so please review and fill in the blanks so I can add and update it.

dotpointer committed:
Updating information about Bower, Yarn, staging, alpha and beta.

docs/20-glossary.rst, moving and updating Bower, updating Git, adding Yarn
docs/app/10-boot.rst, updating details about cz.config.js, adding details about cz.version.js
docs/app/30-states.rst, adding alpha environment, removing staging environment
docs/frontend/10-repo-setup.rst, updating pre-commit hook, updating development tools and dependencies
docs/frontend/30-working-with-the-code.rst, updating backend information, web server starting, login to alpha and beta
docs/intro/10-development-environment.rst, updating links so they don't conflict, removing gulp, bower and polymer-cli section, adding section about yarn

JockeCK commented on docs/20-glossary.rst:
```diff
@@ -47,7 +47,7 @@ help others.
   `Gulp <http://gulpjs.com/>`_
```
Do we still need info about gulp, can't we just remove it completely instead?

dotpointer replied:
I have now removed texts about gulp. Replaced gulp serve with yarn start where I found possible.

dotpointer commented:
Comment from plequang:

> I think that bower is still needed. Actually we should follow this https://www.polymer-project.org/2.0/docs/tools/polymer-cli#install, replacing npm with yarn

dotpointer committed:
Review changes, removing Gulp and replacing it with Yarn after review from jockeck, adding back bower installation after review from plequang

docs/20-glossary.rst, removing gulp by review from jockeck
docs/frontend/30-working-with-the-code.rst, replacing gulp with yarn after review from jockeck
docs/intro/10-development-environment.rst, adding back information about bower install after review from plequang

dotpointer commented:
Installation instructions for bower are added back as indirectly requested by plequang.

JockeCK reviewed (approved):
LGTM
